server/src/user/user.service.ts

```diff
@@ -64,4 +74,40 @@ export class UserService {
       throw error;
     }
   }
+
+  async updateUser(userId: number, dto: UpdateUserDto) {
+    try {
+      const user = await this.prisma.user.update({
+        where: {
+          id: userId,
+        },
+        data: dto,
+      });
+
+      if (!user) {
+        throw new NotFoundException(`User with ID ${userId} not found.`);
+      }
+
+      const token = await this.auth.generateJwtToken(
+        user.id,
+        user.name,
+        user.email,
+        user.icon,
+      );
+
+      return {
+        token,
+        user,
```

Comment:
Delete the hashed password! We don't want to return the secure information

Reply:
Oops! I fixed it. 
342d09ff8aa72a79e494a1b985d1ad5f95a9c9a7